shell/platform/common/client_wrapper/basic_message_channel_unittests.cc

```diff
@@ -20,23 +21,38 @@ class TestBinaryMessenger : public BinaryMessenger {
   void Send(const std::string& channel,
             const uint8_t* message,
             const size_t message_size,
-            BinaryReply reply) const override {}
+            BinaryReply reply) const override {
+    send_called_ = true;
+    int length = static_cast<int>(message_size);
+    last_message_size_ = length;
+    last_message_ =
+        std::vector<uint8_t>(message, message + length * sizeof(uint8_t));
+  }
 
   void SetMessageHandler(const std::string& channel,
                          BinaryMessageHandler handler) override {
     last_message_handler_channel_ = channel;
     last_message_handler_ = handler;
   }
 
+  bool send_called() { return send_called_; }
+
   std::string last_message_handler_channel() {
     return last_message_handler_channel_;
   }
 
   BinaryMessageHandler last_message_handler() { return last_message_handler_; }
 
+  std::vector<uint8_t> last_message() { return last_message_; }
+
+  int last_message_size() { return last_message_size_; }
+
  private:
+  mutable bool send_called_ = false;
   std::string last_message_handler_channel_;
   BinaryMessageHandler last_message_handler_;
+  mutable std::vector<uint8_t> last_message_;
+  mutable int last_message_size_;
```

Comment:
` = 0`

Reply:
Although, why do we need this variable, rather than just checking the size of `last_message_`?